src/browser/api_protocol/api_handlers/system.js

```diff
@@ -80,6 +81,19 @@ function SystemApiHandler() {
         return e !== undefined && e.constructor === Error;
     }
 
+    function readRegistryValue(identity, message, ack, nack) {
+        const { payload: { rootKey, subkey, value } } = message;
+        const response = System.readRegistryValue(rootKey, subkey, value);
+
+        if (_.isEmpty(response)) {
+            nack(new Error('There was an error reading the registry.'));
```

Comment:
is there anywhere in the runtime where we can get the actual error that caused an empty read?